llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -199,93 +170,151 @@ groupEntryPointsByKernelType(ModuleDesc &MD,
   return EntryPointGroups;
 }
 
-// Represents a call graph between functions in a module. Nodes are functions,
-// edges are "calls" relation.
-class CallGraph {
+// Represents "dependency" or "use" graph of global objects (functions and
+// global variables) in a module. It is used during device code split to
+// understand which global variables and functions (other than entry points)
+// should be included into a split module.
+//
+// Nodes of the graph represent LLVM's GlobalObjects, edges "A" -> "B" represent
+// the fact that if "A" is include into a module, then "B" should be included
+// as well.
+//
+// Examples of dependencies which are represented in this graph:
+// - Function FA calls function FB
+// - Function FA uses global variable GA
+// - Global variable GA references (intiialized with) function FB
+// - Function FA stores address of a function FB somewhere
+// - Function FA uses global variable GB
+//
+// The following cases are treated as dependencies between global objects:
+// 1. Global object A is used within by a global object B in any way (store,
```

Comment:
Does this include: Global Object A is used in Global Object B's initializer?

Thanks

Reply:
Yes, one global being initializer for another global constitutes as a use within LLVM and therefore we will also treat it as a dependency. I've added corresponding LIT test, which is based on actual example we have with virtual functions, where global variable with `vtable` references some functions, see ece594bf127a9ba28052ab30c332d846a04f4044